src/DynamoPackages/PackageAssemblyLoadContext.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Dynamo.PackageManager
+{
+    internal class PkgAssemblyLoadContext : AssemblyLoadContext
+    {
+        private readonly string RootDir;
+        private IEnumerable<FileInfo> pkgAssemblies = null;
+        public PkgAssemblyLoadContext(string name, string pkgRoot, bool unloadable = true) : base(name, unloadable)
+        {
+            this.RootDir = pkgRoot;
+        }
+
+        protected override Assembly Load(AssemblyName assemblyName)
+        {
+            pkgAssemblies ??= new DirectoryInfo(RootDir).EnumerateFiles("*.dll", new EnumerationOptions() { RecurseSubdirectories = true });
+
+            var targetAssemName = assemblyName.Name + ".dll";
+            var targetAssembly = pkgAssemblies.FirstOrDefault(x => x.Name == targetAssemName);
+            if (targetAssembly != null)
+            {
+                return LoadFromAssemblyPath(targetAssembly.FullName);
+            }
+            return null;
+        }
+
+        protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
```

Comment:
Does the fact that you can override this function mean that you can also load unmanaged assemblies in isolation?

Reply:
Sadly no, you can only override the search algorithm